File: 2022/day_01/main.R
# Import the readr library
setwd("~/github/advent_of_code/2022/day_01")

verbose = FALSE

# read_lines() to read one line at a time
myData = readLines("input.txt", n = -1)

this_elf <- 0
max_calories <- 0

for (x in myData) {

  if (x == "") {

    if (verbose) {
      print(this_elf)
    }

    max_calories = max(c(max_calories, this_elf))
    this_elf <- 0

  } else {

    calories <- strtoi(x)
    this_elf <- this_elf + calories

  }

}

print(sprintf("max calories is %i", max_calories))

File: 2022/day_01/main.py
from pathlib import Path

from rich import print

verbose = False

input_file = Path(__file__).parent.joinpath("input.txt")

with input_file.open() as f:
    input_data = f.read()

elves = []
this_elf = []
for row in input_data.splitlines():

    if row == "":
        elves.append(sum(this_elf))
        if verbose:
            print(this_elf)
            print(sum(this_elf))
        this_elf = []

    else:
        this_elf.append(int(row))

if verbose:
    print(elves)

print(f"max value is {max(elves)}")
